src/reducers-actions/trade.ts

```diff
@@ -0,0 +1,30 @@
+import { TokenDetails } from '@gnosis.pm/dex-js'
+
+export interface TradeState {
+  price: string | null
+  sellAmount: string | null
+  sellToken: Required<TokenDetails> | null
+  buyToken: Required<TokenDetails> | null
+  validFrom: string
+  validUntil: string
+}
+
+export function updateTradeState<K extends keyof TradeState>(
```

Comment:
What is this function, this should be an action constructor, something like:
```ts
const updateTradeState = (payload:Partial<TradeState>): YourAction => ({type: YOUR_TYPE, payload})
```

But if you go without actions in reducer (which is a completely valid approach), essentially making it the same as `setState`, then you don't even need `updateTradeState`. Just `dispatch(newTradeState_partial)`

Reply:
Now about that _completely valid approach_.
Won't work nicely every time. Why? Because the dispatch you use in react components is a `GlobalState.dispatch`. Whatever you give it goes through the `combinedReducer` before it gets to `trade::reducer`. It only works here because every other reducer checks for `action.type` and `updateTradeState` returns without `type`.

If you had any other reducer that operated without types, that other reducer would react to TradeState and spread it over whatever it spreads over, like butter :custard:  over bread :bread: , but wrong butter :fish_cake:  over wrong bread :baguette_bread: 

So do proper actions instead.

Reply:
Hmmm, so that's why. If you check the commit history, I started cutting it off bit by bit to see how far the action boilerplate was needed. Turns out it wasn't, but now you explained why it's there in the first place :)
Will roll back